cmd/native-btc/relayer.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"bytes"
+	"encoding/hex"
+	"os"
+	"time"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/gonative-cc/relayer/bitcoin"
+	"github.com/gonative-cc/relayer/dal"
+	"github.com/rs/zerolog/log"
+)
+
+// Config holds the configuration parameters for the Relayer.
+type Config struct {
+	DatabasePath string `json:"databasePath"`
+	BitcoinNode  string `json:"bitcoinNode"`
+}
+
+// Relayer broadcasts pending transactions from the database to the Bitcoin network.
+type Relayer struct {
+	config       Config
+	db           *dal.DB
+	btcClient    bitcoin.Client
+	shutdownChan chan struct{}
+}
+
+// NewRelayer creates a new Relayer instance with the given configuration.
+func NewRelayer(config Config, db *dal.DB) (*Relayer, error) {
+	var err error
+	if db == nil {
+		db, err = dal.NewDB(config.DatabasePath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	connCfg := &rpcclient.ConnConfig{
+		Host:         os.Getenv("BTC_RPC"),
+		User:         os.Getenv("BTC_RPC_USER"),
+		Pass:         os.Getenv("BTC_RPC_PASS"),
+		HTTPPostMode: true,
+		DisableTLS:   false,
+	}
+
+	client, err := rpcclient.New(connCfg, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Relayer{
+		config:       config,
+		db:           db,
+		btcClient:    client,
+		shutdownChan: make(chan struct{}),
+	}, nil
+}
+
+// Start starts the relayer's main loop to broadcast transactions.
+func (r *Relayer) Start() {
+	go func() {
+		<-r.shutdownChan
+		err := r.db.Close()
+		if err != nil {
+			log.Err(err).Msg("Error closing database connection")
+		}
+		r.btcClient.Shutdown()
+	}()
+
+	for {
```

Comment:
* Use ticker: https://gobyexample.com/tickers
* don't use go routine here to stop the client.

we should have something like this:

```go
ticker := time.NewTicker(time.Second * 10)
for {
	select {
	case <- r.shutdownChan:
		r.btcClient.Shutdown()
        return nil
	case <-ticker.C:
       if err := r.ProcessPendingTxs(); err != nil {
           // TODO: check type of error to decide if we shutdown or not 

           r.btcClientShutdown()
           ...
           return err
       }
	}
}
```